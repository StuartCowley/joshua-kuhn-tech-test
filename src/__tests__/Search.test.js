import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Search />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("button exists", () => {
    render(<Search />);

    const searchButton = screen.getByRole("button", { name: /Search/i });

    expect(searchButton).toBeInTheDocument();
  });

  it("updates input value correctly", () => {
    render(<Search setSearchResults={() => {}} />);

    const input = screen.getByRole("textbox");

    fireEvent.change(input, { target: { value: "test" } });

    expect(input.value).toBe("test");
  });
});
