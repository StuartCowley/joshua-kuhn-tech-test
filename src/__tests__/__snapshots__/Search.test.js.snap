// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <div
    data-testid="search-component"
  >
    <form
      class="search-form"
    >
      <input
        class="search-input"
        type="text"
      />
      <button
        class="search-button"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
